flink/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory

```diff
@@ -13,4 +13,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-org.apache.iceberg.flink.FlinkDynamicTableFactory
\ No newline at end of file
+org.apache.iceberg.flink.FlinkDynamicTableFactory
+org.apache.iceberg.flink.FlinkCatalogFactory
```

Comment:
Should we remove this file [org.apache.flink.table.factories.TableFactory)](https://github.com/apache/iceberg/blob/master/flink/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory) if we plan to upgrade the flink version to 1.13.2 directly ?

Reply:
If we only support 1.13, then is ok to remove it. But if we want to be compatible with 1.12, `FlinkCatalogFactory` need to be treated as `TableFactory` because `Context` class is missing in 1.12 and without implementing `createCatalog(Context context)` will throw exception when we place `FlinkCatalogFactory` in [ org.apache.flink.table.factories.Factory](https://github.com/apache/iceberg/blob/master/flink/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory)